Use redux in components

fuadmoin opened the pull request:
In this branch I: 

- [ ] imported the booksSlice into home.js
- [ ] created event listeners to the form
- [ ] made accepted author and title to be added to the books array
- [ ] added a remove button
- [ ] populated the initial state with the given data
- [ ] added add and remove functionality

fuadmoin committed:
Use redux in components

Zuhaib042 reviewed (changes requested):
Hi @fuadmoin 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
![](https://media.giphy.com/media/31Q36EHJ7C5ZoIMw79/giphy.gif)

## Highlights
✅ Descriptive PR
✅ Professional Readme

## Required Changes ♻️
- [ ] You have done extremely well in this milestone and all the required functionalities are working but there is one issue. As described in the requirements, you need to create two separate components, one for adding a new book and one for removing a book. Kindly consider refactoring your code and adding two separate components for adding a book and removing a book. following are the requirements:
  -  Add a <Button> component, which includes:
  An event handler that adds a book to the books array (with attributes id, title and author)
  -  Add a <Button> component, which includes:
  An event handler that removes a book from the books array (by id)

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

fuadmoin committed:
Make required changes

I created addBooks and removeBook components and used them to add and remove the book from the array.

didierganthier commented on src/components/removeBook.js:
```diff
@@ -0,0 +1,31 @@
+  })).isRequired,
+  removeBook: PropTypes.func.isRequired,
+};
+export default RemoveBooks;
```
[OPTIONAL] Well done overall 💯 although the component name should reflect the task the component is accomplishing, this is not only removing a book, so I'd suggest breaking this down into 2 components:
 - `BookList` (to show the list of `Book`)
 - `Book` (to show a single book, it will contain the `removeBook` function)

Also, this could be renamed to BookList as this is showing a list of Books. 📚 if you need any more explanation please tag me in this thread. 🤓 

❓ This is OPTIONAL as the changes you made were requested by the previous reviewer

didierganthier commented on src/components/addBooks.js:
```diff
@@ -0,0 +1,37 @@
+import React, { useState } from 'react';
```
[OPTIONAL] Great job once again 💯 but I'd suggest better naming for your components. Ultimately, it's up to you to decide what naming conventions you want to use. However, there are a few different approaches that gained popularity among React developers.

In React applications, there are 3 most common naming conventions:

- Camel case for file names, and pascal case for component names
- Kebab case for file names, and pascal case for component names
- Pascal case for both file names, and component names

As you can see, there is one thing that's pretty much set in stone. And that's the convention for component names. A component in React should always use a pascal case. E.g, `App`, `AppContainer` or `AwesomeAppContainer`.

didierganthier reviewed (approved):
Hi @fuadmoin ,

Congratulations! 🎉 Your project is complete and ready to be merged! 🚀

<img src="https://media.giphy.com/media/l9CSJHK88kMTe/giphy.gif" />

## Highlights
- Descriptive PR ✔️ 
- Linters passing ✔️ 
- Correct Gitflow ✔️ 

## Optional Suggestions 💡
- While these comments with the [OPTIONAL] prefix won't prevent the approval of this pull request, I strongly recommend taking them into account as they can improve your code. Some of them were missed by the previous reviewer and addressing them will really enhance your application.

Cheers and Happy coding!👏👏👏

If you have any questions or need clarification on anything, feel free to leave a comment in the pull request thread and don't forget to tag me so I'll receive the notification.

------
As per the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project. If you feel that the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s). 

Once again, congratulations on completing this project! 🎉

fuadmoin committed:
Make suggested optional changes
